fraction.rb

```diff
@@ -1,17 +1,64 @@
 class Fraction
   def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+    if ! numerator.is_a?(Integer) && denominator.is_a?(Integer)
+      # TODO: Proper execption for this issue
+      # TODO: Handle case when one or both parameters are not int but could be easily transformed to int
+      raise 'these are supposed to be numbers: '+numerator.inspect+' '+ denominator.inspect
     end
     @numerator = numerator
     @denominator = denominator
   end
+
+  def numerator
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def simplify
+
+=begin
+     if @numerator > @denominator then b=@denominator else b=@numerator
+    end
+    end
+
+    while b>1
+     if (@numerator%b == 0 and @denominator%b == 0)
+       @denominator = @denominator /b
+       @numerator = @numerator /b
+     end
+      b=b-1
+=end
+    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
+  end
+
+  def view
+      puts "#{@numerator}/#{@denominator}"
+  end
+
+  def +(a)
+    Fraction.new(@numerator*a.denominator+a.numerator*@denominator, @denominator*a.denominator).simplify
+  end
+
+  def *(a)
+    Fraction.new(@numerator = @numerator*a.numerator, @denominator = @denominator*a.denominator).simplify
```

Comment:
Do not create such long lines, it's hard to read. It's just bad style.
Better to do something like:
```ruby
num, den = @numerator * a.numerator, @denominator * a.denominator
Fraction.new(num, den).simplify
```

And be aware about mutating existing data.
Try to call smth like:
```ruby
f1 = Fraction.new(2, 3)
f2 = Fraction.new(5, 7)
f3 = f1 * f2
f1.view  # => 10/21
```
Any thoughts why `f1` was changed too?